src/storage/mod.rs

```diff
@@ -686,6 +689,124 @@ impl<E: Engine, L: LockManager> Storage<E, L> {
         }
     }
 
+    pub fn scan_lock(
+        &self,
+        ctx: Context,
+        max_ts: TimeStamp,
+        start_key: Option<Key>,
+        end_key: Option<Key>,
+        limit: usize,
+    ) -> impl Future<Output = Result<Vec<LockInfo>>> {
+        const CMD: CommandKind = CommandKind::scan_lock;
+        let priority = ctx.get_priority();
+        let priority_tag = get_priority_tag(priority);
+        let concurrency_manager = self.concurrency_manager.clone();
+
+        let res = self.read_pool.spawn_handle(
+            async move {
+                if let Ok(start_key) = start_key
+                    .as_ref()
+                    .map(|k| k.to_raw())
+                    .unwrap_or_else(|| Ok(vec![]))
+                {
+                    let mut key = vec![];
+                    if let Some(end_key) = &end_key {
+                        if let Ok(end_key) = end_key.to_owned().into_raw() {
+                            key = end_key;
+                        }
+                    }
+                    tls_collect_qps(ctx.get_region_id(), ctx.get_peer(), &start_key, &key, false);
+                }
+
+                KV_COMMAND_COUNTER_VEC_STATIC.get(CMD).inc();
+                SCHED_COMMANDS_PRI_COUNTER_VEC_STATIC
+                    .get(priority_tag)
+                    .inc();
+
+                let command_duration = tikv_util::time::Instant::now_coarse();
+
+                concurrency_manager.update_max_ts(max_ts);
+                // TODO: Use better error handling here
```

Comment:
Could you describe it more in detail? Do you mean returning error will make this round of GC fail, so a better way is to keep these locks and scan some more locks to reach the limit?

Reply:
How bout collecting all locks in the `concurrency_manager` and merging it with the result of `scan_lock`?

Reply:
Is it necessary? How do  you think @sticnarf 
and how about Green  GC mode?

Reply:
OK. It's nearly impossible to happen.